### ISSUE TITLE
docker 1.12.3 experimental REST API - Failed to create overlay network

### ISSUE THREAD

**Issue by yaakov-berkovitch** [docker/docker#29032]
**Description**

I'm trying to create an overlay network using the REST API and get reject.

**Steps to reproduce the issue:**
Run the following curl command:
curl -v --unix-socket /var/run/docker.sock -H "Content-Type: application/json" -X POST -d '{"Name": "my_network","driver":"overlay"}' http://localhost/networks/create

**Describe the results you received:**
An error is returned:
HTTP/1.1 500 Internal Server Error
{"message":"rpc error: code = 3 desc = driver name: if driver is specified name is required"}

**Describe the results you expected:**
I expected to get status = 201 and the network ID of the created network

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker version`:**

```
docker version
Client:
 Version:      1.12.3
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   6b644ec
 Built:        
 OS/Arch:      linux/amd64
 Experimental: true

Server:
 Version:      1.12.3
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   6b644ec
 Built:        
 OS/Arch:      linux/amd64
 Experimental: true

```

**Output of `docker info`:**

```
NETWORK ID          NAME                   DRIVER              SCOPE
609a93fe606c        bridge                 bridge              local               
58c7c60882d8        docker_gwbridge        bridge              local               
actn9n8dx0cg        epic_default_network   overlay             swarm               
266d485c56a6        host                   host                local               
1i4ommvfe3fw        ingress                overlay             swarm               
3966d2f76b3a        isolated_nw            bridge              local               
fe1914377ed3        none                   null                local               
[Thu Dec 01 18:45:44 lso@ilptlrvlso05:~ ] $ docker version
Client:
 Version:      1.12.3
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   6b644ec
 Built:        
 OS/Arch:      linux/amd64
 Experimental: true

Server:
 Version:      1.12.3
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   6b644ec
 Built:        
 OS/Arch:      linux/amd64
 Experimental: true
[Thu Dec 01 19:09:10 lso@ilptlrvlso05:~ ] $ docker info
Containers: 7
 Running: 3
 Paused: 0
 Stopped: 4
Images: 14
Server Version: 1.12.3
Storage Driver: devicemapper
 Pool Name: docker-8:1-109052085-pool
 Pool Blocksize: 65.54 kB
 Base Device Size: 10.74 GB
 Backing Filesystem: xfs
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 4.391 GB
 Data Space Total: 107.4 GB
 Data Space Available: 50.17 GB
 Metadata Space Used: 6.246 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.141 GB
 Thin Pool Minimum Free Space: 10.74 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 WARNING: Usage of loopback devices is strongly discouraged for production use. Use `--storage-opt dm.thinpooldev` to specify a custom block storage device.
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.107-RHEL7 (2016-06-09)
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: overlay bridge null host
Swarm: active
 NodeID: en0eltwaqfwjcamt3gjc0v6dq
 Is Manager: true
 ClusterID: er9u6a34ae3bgxjqsp33q6yud
 Managers: 1
 Nodes: 1
 Orchestration:
  Task History Retention Limit: 5
 Raft:
  Snapshot Interval: 10000
  Heartbeat Tick: 1
  Election Tick: 3
 Dispatcher:
  Heartbeat Period: 5 seconds
 CA Configuration:
  Expiry Duration: 3 months
 Node Address: 147.234.150.121
Runtimes: runc
Default Runtime: runc
Security Options: seccomp
Kernel Version: 3.10.0-229.14.1.el7.x86_64
Operating System: CentOS Linux 7 (Core)
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 7.798 GiB
Name: ilptlrvlso05
ID: B2DQ:L3B4:PRPZ:ZLQI:7JDR:4AKA:K7LZ:Q2VI:UKOH:B3LN:WKKD:JLEO
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: true
Insecure Registries:
 globe:5000
 127.0.0.0/8
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Comment by allencloud** [comment]
I experienced this error as well. @yaakov-berkovitch 
In fact, when you hope to create an "overlay" network via swarm mode in 1.12.x, you should also pass a parameter of 
```
"IPAM":{
    "Driver": "default",
    "Options": {
      "foo": "bar"
    }
  },
```
Here "default" is necessary in swarmkit validation.
It is different from other network creations, such "bridge" which do not need `IPAM` at all.

While luckily, in docker 1.13.x, we have fixed this bug, we do not have to pass `IPAM`.

Good luck 😄

**Comment by thaJeztah** [comment]
Yes, this is a duplicate of https://github.com/docker/docker/issues/25735, and is fixed for 1.13, but @allencloud's suggestion should work

**Comment by yaakov-berkovitch** [comment]
Thanks for the so quick answer - I didn't see that same bug was already reported.
